exporter.go

```diff
@@ -159,3 +167,91 @@ func coldStartRequest(target string) error {
 	_, err := client.Get(target)
 	return err
 }
+
+// overwriteModuleParams allows to specify some overwrites for the http prober:
+//
+// http_valid_status_codes
+// Overwrite the valid HTTP status codes of the request being made by defining comma
+// separated status codes. You can also use a shortcut like 2xx for the range of 200-299.
+//
+// http_expect_regexp
+// Define a regular expression which will be matched against the response body. If it matches
+// the probe will be marked as successful.
+//
+// http_fail_on_regexp
+// Define a regular expression which will be matched against the response body. If it matches
+// the probe will be marked as failed.
+//
+// http_basic_auth_username
+// Define a username which will be used for basic auth in the request being made
+//
+// http_basic_auth_password
```

Comment:
This requires including the basic auth password in a query parameter which is considered a bad practice since it will show up in the logs. Maybe we could just pass through the `Authorization: Basic` header for basic auth?

Reply:
Interesting and good idea...will see how this can be achieved.

Reply:
So currently it does not seem to be possible to change headers in prometheus scrape configs (in the relabel_configs). Prometheus maintainers decided against that (https://github.com/prometheus/prometheus/issues/1724). This means that the given basic auth parameters can only be statically given in the scrape config (https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config), but not changed by annotations or labels on resources. Will check if blackbox-exporter could be used by a serviceMonitor (prometheus-operator) as then basic auth credentials could be defined.

Reply:
I checked and currently you can not use a prometheus-operator ServiceMonitor with a ExternalName service in kubernetes (https://github.com/coreos/prometheus-operator/issues/218) as prometheus-operator uses Endpoint resources to determine the target address of the exporter. There are no Endpoint objects being created for ExternalName services. One could create an Endpoint object manually, but as Endpoint objects only support IP addresses and no DNS names, this would be kinda flaky. I will remove the basic auth support for now from the PR as I don't see a current possibility for a secure configuration.